### PR TITLE
CLOUDP-67149: mongocli atlas cluster(s) search index(es) create

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-67149](https://jira.mongodb.org/browse/CLOUDP-67149)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```zsh
mongocli atlas clusters search create myIndex --clusterName myDemo3 --db test3 --collection test --dynamic 
{
  "analyzer": "lucene.standard",
  "collectionName": "test",
  "database": "test3",
  "indexID": "5f118d60a57e107028ab522b",
  "mappings": {
    "dynamic": true
  },
  "name": "myIndex",
  "searchAnalyzer": "lucene.standard"
}
```

### CODE REVIEW

File: internal/cli/atlas/search/create.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		}
+		fields[f[0]] = atlas.IndexField{
+			Analyzer: f[1],
+			Type:     f[2],
+		}
+	}
+	return fields, nil
+}
+
+// mongocli atlas cluster(s) search|fts index(es) create <indexName> [--clusterName clusterName][--db dbName][--collection collection][--dynamic][--field field][--projectId projectId]
```
Comment (robcarlan-mlab) [inline internal/cli/atlas/search/create.go:101]: [np] `[--clusterName clusterName][--db dbName][--collection collection]` -> `--clusterName clusterName --db dbName --collection collection` as they are required?

File: internal/cli/atlas/search/create_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	defer ctrl.Finish()
+
+	opts := &CreateOpts{
+		store: mockStore,
+	}
+
+	request, err := opts.newSearchIndex()
+	if err != nil {
+		t.Fatalf("newOnlineArchive() unexpected error: %v", err)
```
Comment (robcarlan-mlab) [inline internal/cli/atlas/search/create_test.go:39]: copy/paste

File: internal/cli/atlas/search/create.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas/search/create.go]: I think this approach in general needs a revamp, to begin with is not appropriate go doc, it should start with the method name I started it as a way to remember what flags I needed as I coded but let me know what you think of what I've done now

**Comment by gssbzn** [comment]
evergreen retry

**Pull request merged**